app/models/geographic_item.rb

```diff
@@ -121,91 +138,165 @@ class GeographicItem < ApplicationRecord
+        self.select("ST_Union(#{GeographicItem::GEOMETRY_SQL.to_sql}) as collection")
           .where(id: geographic_item_scope.pluck(:id))
       end
 
-      def st_collect(geographic_item_scope)
-        GeographicItem.select("ST_Collect(#{GeographicItem::GEOMETRY_SQL.to_sql}) as collection")
-          .where(id: geographic_item_scope.pluck(:id))
+      # True for those shapes that are within (subsets of) the shape_sql shape.
+      def within_sql(shape_sql)
+        'ST_Covers(' \
+          "#{shape_sql}, " \
+          "#{GeographicItem::GEOMETRY_SQL.to_sql}" \
+        ')'
+      end
+
+      # Note: !! If the target GeographicItem#id crosses the anti-meridian then
+      # you may/will get unexpected results.
+      def within_union_of_sql(*geographic_item_ids)
+        within_sql("#{
+          self.items_as_one_geometry_sql(*geographic_item_ids)
+        }")
+      end
+
+      def within_union_of(*geographic_item_ids)
+        where(within_union_of_sql(*geographic_item_ids))
+      end
+
+      # True for those shapes that cover the shape_sql shape.
+      def covering_sql(shape_sql)
+        'ST_Covers(' \
+          "#{GeographicItem::GEOMETRY_SQL.to_sql}, " \
+          "#{shape_sql}" \
+        ')'
+      end
+
+      # @return [Scope] of items covering the union of geographic_item_ids;
+      # does not include any of geographic_item_ids
+      def covering_union_of(*geographic_item_ids)
+        where(
+          self.covering_sql(
+            self.items_as_one_geometry_sql(*geographic_item_ids)
+          )
+        )
+        .not_ids(*geographic_item_ids)
+      end
+
+      # @param [Integer] geographic_item_id
+      # @param [Integer] buffer can be positive or negative, in meters for the
+      #                  default geographic case
+      # @param [Boolean] geometric: whether the buffer should be created using
+      #                  geographic (default) or geometric coordinates
+      # @return [RGeo::Polygon or RGeo::MultiPolygon]
+      #   The buffer of size `buffer` around geographic_item_id
+      def st_buffer_for_item(geographic_item_id, buffer, geometric: false)
+        geometric = geometric ? '::geometry' : ''
+
+        GeographicItem.select(
+          'ST_Buffer(' \
+            "#{GeographicItem::GEOGRAPHY_SQL}#{geometric}, " \
+            "#{buffer}" \
+          ') AS buffer'
+        )
+        .where(id: geographic_item_id)
+        .first.buffer
       end
 
-      # @return [GeographicItem::ActiveRecord_Relation]
-      # @params [Array] array of geographic area ids
-      def default_by_geographic_area_ids(geographic_area_ids = [])
-        GeographicItem.
-          joins(:geographic_areas_geographic_items).
-          merge(::GeographicAreasGeographicItem.default_geographic_item_data).
-          where(geographic_areas_geographic_items: {geographic_area_id: geographic_area_ids})
+      def st_distance_sql(shape_sql)
```

Comment:
This reminds me.  I think there is going to be a cleaner way to define these using Arel definitions.